platforms/c-mcpu/evaluator/eval_agent/eval_agent.py

```diff
@@ -204,8 +204,8 @@ def generate_source_file(kernel_code):
 def build_source_file():
     try:
         if platform.system() == 'Linux':
-            cmd = ['g++', 'main.cpp', '-omain', '-std=c++11', '-lpthread', '-O3', '-march=native']
-            output = subprocess.check_output(cmd)
+            cmd = ['timeout', '120s', 'g++', 'main.cpp', '-omain', '-std=c++11', '-lpthread', '-O3', '-march=native']
+            output = subprocess.check_output(cmd, timeout=120)
         else:

```

Comment:
Do you really need to limit the timeout in 2 places? I think the timeout command in command line is not necessary.

Reply:
Sorry for the late response.  I found the timeout setting in check_output will not work sometimes, so I added "timeout" into cmd.